packages/stark-ui/src/modules/session-ui/effects/session-timeout-warning.effect.spec.ts

```diff
@@ -0,0 +1,185 @@
+			]
+		}).compileComponents();
+	}));
+
+	beforeEach(() => {
+		component = TestBed.get(StarkSessionTimeoutWarningDialogEffects);
+		mockSessionService = TestBed.get(STARK_SESSION_SERVICE);
+		mockDialogService = TestBed.get(MatDialog);
+		mockSessionUiConfig = TestBed.get(STARK_SESSION_UI_CONFIG);
+	});
+
+	describe("on initialization", () => {
+		it("should set internal component properties", () => {
+			expect(component.sessionService).not.toBeNull();
+			expect(component.sessionService).toBeDefined();
+			expect(component.starkSessionUiConfig).not.toBeNull();
+			expect(component.starkSessionUiConfig).toBeDefined();
+		});
+	});
+
+	describe("On StarkSessionTimeoutWarning$", () => {
+		it("Should open a dialog when the timeout countdown begins", () => {
+			const afterClosedResult: string = "keep-logged";
+			const afterClosed$: Subject<string> = new Subject();
+
+			(<Spy>mockDialogService.open).and.returnValue({
+				afterClosed: () => {
+					return afterClosed$;
+				}
+			});
+
+			const mockObserver: Observer<any> = createSpyObj<Observer<any>>("observerSpy", ["next", "error", "complete"]);
+			const subject: ReplaySubject<any> = new ReplaySubject(1);
+			actions = subject.asObservable();
+
+			(<Spy>mockDialogService.open).calls.reset();
+
+			component.starkSessionTimeoutWarning$().subscribe(mockObserver);
+
+			subject.next(new StarkSessionTimeoutCountdownStart(20));
+
+			expect(mockSessionService.pauseUserActivityTracking).toHaveBeenCalled();
+			expect(mockObserver.next).toHaveBeenCalledTimes(1);
+			expect(mockObserver.error).not.toHaveBeenCalled();
+			expect(mockObserver.complete).not.toHaveBeenCalled();
+			expect(mockDialogService.open).toHaveBeenCalledTimes(1);
+			expect(mockDialogService.open).toHaveBeenCalledWith(StarkSessionTimeoutWarningDialogComponent, { data: 20 });
+
+			expect(mockSessionService.resumeUserActivityTracking).not.toHaveBeenCalled();
+
+			afterClosed$.next(afterClosedResult);
+
+			expect(mockSessionService.resumeUserActivityTracking).toHaveBeenCalledTimes(1);
+		});
+	});
+
+	describe("On StarkSessionTimeoutWarningClose$", () => {
+		it("Should close the dialog when the countdown finishes", () => {
+			const mockObserver: Observer<any> = createSpyObj<Observer<any>>("observerSpy", ["next", "error", "complete"]);
+
+			const subject: ReplaySubject<any> = new ReplaySubject(1);
+			actions = subject.asObservable();
+
+			(<Spy>mockDialogService.open).and.callThrough(); //simulates an open dialog
+
+			component.starkSessionTimeoutWarningClose$().subscribe(mockObserver);
+
+			subject.next(new StarkSessionTimeoutCountdownFinish());
+
+			expect(mockObserver.next).toHaveBeenCalledTimes(1);
+			expect(mockObserver.error).not.toHaveBeenCalled();
+			expect(mockObserver.complete).not.toHaveBeenCalled();
+			expect(mockDialogService.closeAll).toHaveBeenCalledTimes(1);
+		});
+	});
+
+	describe("On ngrxOnRunEffects", () => {
+		it("Shouldn't use the effects", () => {
```

Comment:
Can you add more info in the description... "should stop the effects immediately when the option xxxx is true"